.github/workflows/aderyn.yaml

```
@@ -0,0 +1,36 @@
+name: Aderyn Static Analysis
+
+on:
+  pull_request:
+    branches:
+      - ci/aderyn-analysis
+      - main
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  aderyn:
+    runs-on: blacksmith-2vcpu-ubuntu-2204
+
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Setup Rust
+        uses: actions-rs/toolchain@v1
+        with:
+          toolchain: stable
+
+      - name: Install Aderyn
+        run: cargo install --git https://github.com/Cyfrin/aderyn.git
+
+      - name: Run Aderyn Analysis
+        run: aderyn .
```

maybe specify the output to make it clear?
```suggestion
        run: aderyn --output report.md .
```